GUI - unify on QSplashScreen instead of QMainWindow

lilyinstarlight opened the pull request:
As we discussed, this PR is a follow-up to 7569cfa for testing since QSplashScreen seems to have better cross-platform support and the issue on macOS motivating 10f5fc7 may no longer be a problem

This needs testing on macOS to make sure there is no problematic regression in the splash screen, especially on retina displays

lilyinstarlight committed:
GUI - unify on QSplashScreen instead of QMainWindow

Follow-up to 7569cfa since QSplashScreen has better cross-platform
support and the issue motivating 10f5fc7 may no longer be a problem.

ethancrawford commented:
I've tested this on my macbook - seems to be ok (though I'm not 100% sure what the original problem was).
Here's a gif of the result: the only oddity I notice is that the top left corner of the splash screen is more rounded/cut off than the others.
![splash](https://user-images.githubusercontent.com/10395940/143666054-29f40686-5481-4e8a-b0a6-c29ed9373261.gif)

lilyinstarlight committed:
GUI - move QApplication attributes before instantiation and update preproc guards for consistency

Changes:
* Move QApplication attributes to be set before instantiation per info
  at https://doc.qt.io/qt-5/qcoreapplication.html#setAttribute and
  https://doc.qt.io/qt-5/qt.html#ApplicationAttribute-enum
* `Q_OS_MAC` is deprecated per https://doc.qt.io/qt-5/qtglobal.html#Q_OS_MAC
* `#elif` is not the same comparison as `#ifdef` (use `#if defined()`)
* Use Qt built-in platform defines
* Use forward slash for Windows import per
  https://doc.qt.io/qt-5/qwindowswindowfunctions.html

lilyinstarlight commented:
I've added updates which should fix the corner and potentially other assorted Qt initialization issues. The splash screen probably needs testing on both Windows and macOS now to ensure there are no regressions since I had to remove the `setMask` call and Windows used it prior to 7569cfa

ethancrawford commented:
Looks good on macOS for me 🙂 
![splash](https://user-images.githubusercontent.com/10395940/143773105-433e02ce-9c99-4bcc-9deb-56eaaf756cab.gif)

rbnpi commented:
Works fine on Mac and Ubuntu for me with no distortion of corner. On Pi4 with new RP Bullseye based OS splash screen only shows properly on first launch. Thereafter it appears very briefly just for a fraction of a second just before the main screen shows. If you log out of the OS gui and in again it works on the first subsequent run of SP. The sqashed corner is however also corrected on the Pi4.

samaaron commented:
I think we should merge and try it for a while. If we get any more issues, we can always roll back to the splash-related code as it was at v3.3.1 prior to release of v4.

What does everyone else think?

rbnpi commented:
Merging fine by me.

ethancrawford commented:
Sure, that sounds ok.

lilyinstarlight commented:
I just did another quick force-push to make a small change. Assuming the CI passes this change, I'm also good with merging and rolling back to the old splash code if it causes weird problems that are difficult to find and fix

samaaron commented:
Thanks so much :-)
